src/sql/workbench/parts/jobManagement/browser/notebooksView.component.ts

```diff
@@ -0,0 +1,988 @@
+		let statusBar = jQuery('.part.statusbar').get(0);
+		if (jobsViewToolbar && statusBar) {
+			let toolbarBottom = jobsViewToolbar.getBoundingClientRect().bottom + ACTIONBAR_PADDING;
+			let statusTop = statusBar.getBoundingClientRect().top;
+			this._table.layout(new dom.Dimension(
+				dom.getContentWidth(this._gridEl.nativeElement),
+				statusTop - toolbarBottom));
+		}
+	}
+
+	onFirstVisible() {
+		let self = this;
+		let cached: boolean = false;
+		if (this._notebookCacheObject.serverName === this._serverName && this._notebookCacheObject.notebooks.length > 0) {
+			cached = true;
+			this.notebooks = this._notebookCacheObject.notebooks;
+		}
+
+		let columns = this.columns.map((column) => {
+			column.rerenderOnResize = true;
+			return column;
+		});
+		let options = <Slick.GridOptions<any>>{
+			syncColumnCellResize: true,
+			enableColumnReorder: false,
+			rowHeight: ROW_HEIGHT,
+			enableCellNavigation: true,
+			forceFitColumns: false
+		};
+
+		this.dataView = new Slick.Data.DataView({ inlineFilters: false });
+
+		let rowDetail = new RowDetailView<IItem>({
+			cssClass: '_detail_selector',
+			process: (job) => {
+				(<any>rowDetail).onAsyncResponse.notify({
+					'itemDetail': job
+				}, undefined, this);
+			},
+			useRowClick: false,
+			panelRows: 1,
+			postTemplate: () => '', // I'm assuming these code paths are just never hit...
+			preTemplate: () => '',
+		});
+		this.rowDetail = rowDetail;
+		columns.unshift(this.rowDetail.getColumnDefinition());
+		let filterPlugin = new HeaderFilter<{ inlineFilters: false }>();
+		this._register(attachButtonStyler(filterPlugin, this._themeService));
+		this.filterPlugin = filterPlugin;
+		jQuery(this._gridEl.nativeElement).empty();
+		jQuery(this.actionBarContainer.nativeElement).empty();
+		this.initActionBar();
+		this._table = new Table(this._gridEl.nativeElement, { columns }, options);
+		this._table.grid.setData(this.dataView, true);
+		this._table.grid.onClick.subscribe((e, args) => {
+			let notebook = self.getNotebook(args);
+			self._agentViewComponent.notebookId = notebook.jobId;
+			self._agentViewComponent.agentNotebookInfo = notebook;
+			self._agentViewComponent.showNotebookHistory = true;
+		});
+		this._register(this._table.onContextMenu(e => {
+			self.openContextMenu(e);
+		}));
+
+		if (cached && this._agentViewComponent.refresh !== true) {
+			this.onNotebooksAvailable(null);
+			this._showProgressWheel = false;
+			if (this.isVisible) {
+				this._cd.detectChanges();
+			}
+		} else {
+			let ownerUri: string = this._commonService.connectionManagementService.connectionInfo.ownerUri;
+			this._jobManagementService.getNotebooks(ownerUri).then((result) => {
+				if (result && result.notebooks) {
+					self.notebooks = result.notebooks;
+					self._notebookCacheObject.notebooks = self.notebooks;
+					self.onNotebooksAvailable(result.notebooks);
+				} else {
+					// TODO: handle error
```

Comment:
should we be handling this?

Reply:
Code was taken from jobsView.component.ts 
Maybe I can show a notification that notebook fetch has failed